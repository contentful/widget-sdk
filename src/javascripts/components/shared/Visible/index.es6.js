import PropTypes from 'prop-types';

/**
 * <Visible /> renders children depending on the value of the prop if.
 * Can be used in case if expressions like `{true && expression}` or `{false && expression}`
 * create to much noise that harms readabillity.
 */
const Visible = props => {
  if (props.if) {
    return props.children;
  }
  return null;
};

Visible.propTypes = {
  if: PropTypes.bool
};

export default Visible;
