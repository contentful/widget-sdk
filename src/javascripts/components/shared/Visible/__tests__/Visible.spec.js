import React from 'react';
import Enzyme from 'enzyme';
import Visible from '../index';

describe('Visible', () => {
  it('renders children if condition is truthy', () => {
    const component = Enzyme.mount(<Visible if={true}>Works</Visible>);
    expect(component).toMatchSnapshot();
  });

  it("doesn't render children if condition is falsy", () => {
    const component = Enzyme.mount(<Visible if={false}>Works</Visible>);
    expect(component).toMatchSnapshot();
  });
});
