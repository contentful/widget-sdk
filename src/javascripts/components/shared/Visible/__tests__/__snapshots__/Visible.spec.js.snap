// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Visible doesn't render children if condition is falsy 1`] = `
<Visible
  if={false}
/>
`;

exports[`Visible renders children if condition is truthy 1`] = `
<Visible
  if={true}
>
  Works
</Visible>
`;
